### DISCUSSION TITLE
v0.4.25

### DISCUSSION THREAD

**Discussion by lbbrhzn** [lbbrhzn/ocpp#446] [Announcements]
## Changes

* add authorization list (#437) @lbbrhzn

<hr /><em>This discussion was created from the release <a href='https://github.com/lbbrhzn/ocpp/releases/tag/v0.4.25'>v0.4.25</a>.</em>

**Comment by dreimer1986** [comment]
Interesting idea. That means I can remove the hard coded ID tag and control with the the real NFC ones. :D

Just one small suggestion. (Likely it's planned already anyways).
I read what it can do and it's a great thing to add. BUT Home Assistant tries to get rid of the configuration.yaml where ever possible.
So maybe the auth support should be added to first configuration AND the Configure button in the Web UI instead.
Crazy to add without risking to lose some functionality and I personally have no problem with configs at all.
But HA tries to become sort of useable for the everyday user with zero skills in PC and network architecture or any YAML skills.

**Reply by lbbrhzn** [reply]
Thanks for your feedback. I wanted to get something working quickly first, and then migrate from yaml to gui. It appears that it still does not work though, so I have some more development and testing to do.

An advantage of the yaml file is that the list is persistent: you will not need to add the list again when you add the integration again. I think this will be difficult to do with a config flow or options flow.

**Reply by dreimer1986** [reply]
Tbh, I did not test it yet, still at work.

Yeah I think that is impossible by normal ways. Unless you write these information to a file in /config or /share and give a way to restore from there if the file is available. Hack, but should be working.
